roscala/src/main/scala/coop/rchain/rosette/prim/Number.scala

```diff
@@ -84,18 +84,373 @@ object Number {
+      val n = ctxt.nargs
+      Right(ctxt.argvec.elem.take(n).foldLeft(Fixnum(0)) {
+        case (accum, fixnum: Fixnum) => accum ^ fixnum
+      })
+    }
+  }
+
+  object fxLognot extends Prim {
+    override val name: String = "fx-lognot"
+    override val minArgs: Int = 1
+    override val maxArgs: Int = 1
+
+    @checkTypeMismatch[Fixnum]
+    @checkArgumentMismatch
+    override def fn(ctxt: Ctxt): Either[PrimError, Fixnum] = {
+      val n = ctxt.argvec.elem.head.asInstanceOf[Fixnum]
+      Right(Fixnum(~n.value))
+    }
+  }
+
+  object fxMdiv extends Prim {
+    override val name: String = "fx-mdiv"
+    override val minArgs: Int = 0
+    override val maxArgs: Int = MaxArgs
+
+    @checkTypeMismatch[Fixnum]
+    @checkArgumentMismatch
+    override def fn(ctxt: Ctxt): Either[PrimError, Fixnum] = {
+      val n = ctxt.nargs
+
+      val commonBits = ctxt.argvec.elem.take(n).foldLeft(Fixnum(~0)) {
+        case (accum, fixnum: Fixnum) => accum & fixnum
+      }
+      if (commonBits.value != 0) {
+        Right(Fixnum(1))
+      } else {
+        Right(Fixnum(0))
+      }
+    }
+  }
+
+  object fxCdiv extends Prim {
+    override val name: String = "fx-cdiv"
+    override val minArgs: Int = 2
+    override val maxArgs: Int = 2
+
+    @checkTypeMismatch[Fixnum]
+    @checkArgumentMismatch
+    override def fn(ctxt: Ctxt): Either[PrimError, Fixnum] = {
+      val m = ctxt.argvec.elem.head.asInstanceOf[Fixnum]
+      val n = ctxt.argvec.elem(1).asInstanceOf[Fixnum]
+
+      try {
+        Right(Fixnum(Math.ceil(m.value.toDouble / n.value.toDouble).toInt))
+      } catch {
+        case e: ArithmeticException =>
+          Left(ArithmeticError)
+      }
+    }
+  }
+
+  object fxAsl extends Prim {
+    override val name: String = "fx-asl"
+    override val minArgs: Int = 2
+    override val maxArgs: Int = 2
+
+    @checkTypeMismatch[Fixnum]
+    @checkArgumentMismatch
+    override def fn(ctxt: Ctxt): Either[PrimError, Fixnum] = {
+      val m = ctxt.argvec.elem.head.asInstanceOf[Fixnum]
+      val n = ctxt.argvec.elem(1).asInstanceOf[Fixnum]
+
+      Right(m << n)
+    }
+  }
+
+  object fxAsr extends Prim {
+    override val name: String = "fx-asr"
+    override val minArgs: Int = 2
+    override val maxArgs: Int = 2
+
+    @checkTypeMismatch[Fixnum]
+    @checkArgumentMismatch
+    override def fn(ctxt: Ctxt): Either[PrimError, Fixnum] = {
+      val m = ctxt.argvec.elem.head.asInstanceOf[Fixnum]
+      val n = ctxt.argvec.elem(1).asInstanceOf[Fixnum]
+
+      Right(m >> n)
+    }
+  }
+
+  object fxLsl extends Prim {
+    override val name: String = "fx-lsl"
+    override val minArgs: Int = 2
+    override val maxArgs: Int = 2
+
+    @checkTypeMismatch[Fixnum]
+    @checkArgumentMismatch
+    override def fn(ctxt: Ctxt): Either[PrimError, Fixnum] = {
+      val m = ctxt.argvec.elem.head.asInstanceOf[Fixnum]
+      val n = ctxt.argvec.elem(1).asInstanceOf[Fixnum]
+
+      Right(m <<< n)
```

Comment:
Can you explain why you use `<<<` here instead of `<<`?

Reply:
Yes, you're right. I've mix them up.